### ISSUE TITLE
Usage of renderToStaticMarkup from react-dom/server in react Text component bloats client filesize

### ISSUE THREAD

**Issue by morbidick** [ibitcy/eo-locale#107]
Hi, thank you for this nice and small translation library.

In commit 3c1d0a05c9058e3f06c2d5dffd4d790f344c2c79 `renderToStaticMarkup` from `react-dom/server` was introduced to the Text Component, unfortunately that leads to the inclusion for react-dom-server in the client build and bloating the filesize.

**Comment by appbak3r** [comment]
@pret-a-porter That's actually an issue for vite for example, having `react-dom/server` increases bundle size by 25% and it makes no sense to use the library

**Comment by lubieowoce** [comment]
> Yes it is, but it allows to use React components as parameters for Text component

Right, but that component won't have access to things like context, because it's rendered in isolation. That feels like a pretty big footgun.

**Comment by pret-a-porter** [comment]
Make sense, issue reopened

**Comment by lubieowoce** [comment]
AFAIK the "proper" way to handle it is to make Text return a fragment, and let react handle the rest. So, something like the following:
```js
// messages: { hello: 'Hello, {name}!' }

// a Text like this:
<Text id="hello" name={<SomeComponent />} />

// should output (return) this:
<>{"Hello, "}<SomeComponent />{"!"}</>
```

**Comment by pret-a-porter** [comment]
Unfortunately it does not cover case, when string contains html tags. React fragments does not support `dangerouslySetInnerHTML` and this proposal https://github.com/reactjs/rfcs/pull/129 still in RFC stage for more than two years.

**Comment by lubieowoce** [comment]
yes, html strings would probably need to be parsed into a react tree (a la `react-html-parser`). not ideal, but i don't think there's a way around it if you want to have `<Elements>` work properly. i guess at least its cacheable (the translations don't really change much) so shouldn't be a big perf problem

**Comment by lukawski** [comment]
Hi, did you consider restructuring library shape to enable better tree shaking? I think if we replaced current single module https://github.com/ibitcy/eo-locale/blob/master/packages/react/src/index.tsx with each component/hook living in its own module file, it could help with that. 
Example structure: 
```
lib/
  TranslationsProvider.js   ← no react-dom/server import
  useTranslator.js          ← no react-dom/server import
  Text.js                   ← imports renderToStaticMarkup
  index.js                  ← re-exports from all of the above
```
